app/src/main/java/io/gnosis/safe/ui/safe/selection/SafeSelectionViewModel.kt

```diff
@@ -0,0 +1,85 @@
+package io.gnosis.safe.ui.safe.selection
+
+import android.os.Bundle
+import androidx.navigation.NavDirections
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.R
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import javax.inject.Inject
+
+
+sealed class SafeSelectionState : BaseStateViewModel.State {
+
+    data class SafeListState(
+        val listItems: List<Any>,
+        val activeSafe: Safe?,
+        override var viewAction: BaseStateViewModel.ViewAction?
+    ) : SafeSelectionState()
+
+    data class AddSafeState(
+        override var viewAction: BaseStateViewModel.ViewAction?
+    ) : SafeSelectionState()
+}
+
+class SafeSelectionViewModel @Inject constructor(
+    repositories: Repositories,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SafeSelectionState>(appDispatchers),
+    SafeSelectionAdapter.OnSafeSelectionItemClickedListener {
+
+    private val safeRepository = repositories.safeRepository()
+
+    private val items: MutableList<Any> = mutableListOf()
+    private var activeSafe: Safe? = null
+
+    override fun initialState(): SafeSelectionState =
+        SafeSelectionState.SafeListState(
+            listOf(AddSafeHeader()), null, null
+        )
+
+    fun loadSafes() {
+        safeLaunch {
+            val safes = safeRepository.getSafes()
+            activeSafe = safeRepository.getActiveSafe()
+
+            items.clear()
+            items.add(AddSafeHeader())
+            items.addAll(safes)
+
+            updateState { SafeSelectionState.SafeListState(items, activeSafe, null) }
+        }
+    }
+
+    fun selectSafe(safe: Safe) {
+        safeLaunch {
+            safeRepository.setActiveSafe(safe)
+            updateState { SafeSelectionState.SafeListState(items, safe, null) }
+        }
+    }
+
+    fun addSafe() {
+        safeLaunch {
+            updateState {
+                SafeSelectionState.AddSafeState(
+                    ViewAction.NavigateTo(
+                        object : NavDirections {
+                            override fun getArguments() = Bundle()
+                            override fun getActionId() = R.id.action_to_add_safe_nav
+                        }
```

Comment:
There is no class that matches this action generated by the safeargs dependency?